### DISCUSSION TITLE
Cannot find a compatible Vulkan device

### DISCUSSION THREAD

**Discussion by marlon31415** [carla-simulator/carla#7616] [General]
After updating my nvidia driver version from
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 525.147.05   Driver Version: 525.147.05   CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
```
to 
```
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.161.07             Driver Version: 535.161.07   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
```
it is not possible to start CARLA anymore. After trying to launch it with `$ ./CarlaUE4.sh`, the error 

> Cannot find a compatible Vulkan device that supports surface presentation

occurs. When switching to the former driver version everything works fine again.

I am on Ubuntu 20.04 and using CARLA 0.9.15 with UE4.26.

**Comment by xavisolesoft** [comment]
Seems an issue with vulkan verion. Find here here the required API versions for nvidia driver 535.161.07:
https://docs.nvidia.com/datacenter/tesla/tesla-release-notes-535-161-07/index.html
Could you check if you have Vulkan 1.3 as suggested in the previous link?

**Reply by marlon31415** [reply]
The following information come from `vulkaninfo`:
```
K_LAYER_LUNARG_standard_validation (LunarG Standard Validation Layer) Vulkan version 1.0.131, layer version 1:
VK_LAYER_MESA_device_select (Linux device selection layer) Vulkan version 1.2.73, layer version 1:
VK_LAYER_MESA_overlay (Mesa Overlay layer) Vulkan version 1.1.73, layer version 1:
VK_LAYER_NV_optimus (NVIDIA Optimus layer) Vulkan version 1.3.242, layer version 1:
```
The NVIDIA Optimus layer has the required Vulkan version of 1.3. This corresponds to the GPU I am using for CARLA.

**Reply by shengyu17** [reply]
Hi @marlon31415 ,

i have encountered exactly the same issue like yours with Nvidia driver version 535.183.01, same CUDA version like yours on Ubuntu 22.04 using CARLA 0.9.15 with UE4.26. Have you solved your issued yet? Thanks in advance.
